torch/utils/_sympy/functions.py

```diff
@@ -559,26 +570,43 @@ def eval(cls, base, shift):
 
 
 class MinMaxBase(Expr, LatticeOp):  # type: ignore[misc]
-    def __new__(cls, *args, **assumptions):
+    def __new__(cls, *original_args, **assumptions):
         from sympy.core.parameters import global_parameters
 
-        evaluate = assumptions.pop("evaluate", global_parameters.evaluate)
-        args = (sympify(arg) for arg in args)
+        import torch._dynamo.config as config
 
-        # first standard filter, for cls.zero and cls.identity
-        # also reshape Max(a, Max(b, c)) to Max(a, b, c)
+        evaluate = assumptions.pop("evaluate", global_parameters.evaluate)
+        args = (sympify(arg) for arg in original_args)
 
         if evaluate:
             try:
+                # first standard filter, for cls.zero and cls.identity
+                # also reshape Max(a, Max(b, c)) to Max(a, b, c)
                 args = frozenset(cls._new_args_filter(args))  # type: ignore[assignment]
             except ShortCircuit:
                 return cls.zero  # type: ignore[attr-defined]
-            # remove redundant args that are easily identified
-            args = cls._collapse_arguments(args, **assumptions)
-            # find local zeros
-            args = cls._find_localzeros(args, **assumptions)
 
-        args = frozenset(args)
+            # When this is not none we can bypass _find_localzeros and _collapse_arguments.
```

Comment:
I think for future understandability, the comment here would be good to explain what "unique summation atoms" denotes. In particular, what does it mean when unique summation atoms is None? Why does this condition mean we should skip collapse arguments / find localzeros? Some of these answers can be found in the docstrings of the function call itself, but *this* is the logical place to explain how it all comes together.

Reply:
done